TFNetworkLayer.py

```diff
@@ -9411,3 +9411,344 @@ def get_layer_class_name_list():
   if not _LayerClassDictInitialized:
     _init_layer_class_dict()
   return sorted(_LayerClassDict.keys())
+
+
+class GatherPositionsLayer(LayerBase):
+  """
+  Gathers vectors at the specified positions on time axis.
+  """
+  layer_class = "gather_positions"
+
+  def __init__(self, **kwargs):
+    """
+    Gathers input data at given positions.
+    Output has shape (B,max(num_positions),None) and is zero-padded for each sequence.
+    Inputs: [LayerBase,LayerBase] [input_data,positions]: with
```

Comment:
You should not assume specific meaning in the order or length of `self.sources`. You should rather have specific named separate arguments, like `positions`.